### PR TITLE
Directory Component

### PR DESCRIPTION
**Pull request by leontaolong**
[Trello Card](https://trello.com/c/0Apl4xNE/12-directory-component)

**Directory Component**
- Took over Directory component frontend PR #7 to work on backend and data integration. 
- Completed the entire Directory component except for filtering functionality. 

**Changes made**
- Added Firebase Integration: data fetching, aggregation, and population for all three types of accounts
- Added logic to determine `AccountType` and display data accordingly
- Changed DirectoryCard UI so it displays `availability` for `receiving_agencies` and `members` of `donating_agencies`
- Added `.indexOn` `umbrella` rules in [Firebase](https://console.firebase.google.com/project/uw-foodwaste/database/uw-foodwaste/rules) for DB optimization

**Tests**
- Correct types of agencies displayed based on the current `accountType`
- Correct content displayed for each agency type
- Real-time UI update for database changes

**TODOs**
- [Frontend] UI tweaks so it better matches design doc: especially for __availabilityList__
- Currently used a placeholder image for org logo. Support org logo in the future
- Filtering functionality Implementation

**Next Steps**
- This component is temporarily considered completed for the pilot.

**Screenshots**
- Displays availability for __receiving agencies__:
screen shot 2018-04-06 at 12 18 31 am

-  Displays members for __donating agencies__ (for testing purposes, DAs won't be displayed if current `accountType` is `donating_agency_member`):
screen shot 2018-04-06 at 12 18 44 am

screen shot 2018-04-06 at 12 18 05 am

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
             "experimentalObjectRestSpread": true,
             "jsx": true
         },
+        "ecmaVersion": 2017,
```
Comment (leontaolong) [inline .eslintrc.json]: Enable ES6 [async/await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await) syntax, which is currently supported by all browsers.

**Comment by leontaolong** [comment]
- Forward Integration
- DirectoryFilter Removed

File: src/PageContent/Directory/DirectoryPage.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        this.fetchOrgs();
+    }
+
+    componentWillUnmount() {
+        // detach listeners
+        db.ref('accounts').orderByChild('umbrella').equalTo(this.state.umbrellaId).off();
+        db.ref('donating_agencies').orderByChild('umbrella').equalTo(this.state.umbrellaId).off();
+    }
+
+    async getUmbrellaId() {
```
Comment (arielin3) [inline src/PageContent/Directory/DirectoryPage.js]: we added `umbrella` field to `DONATING_AGENCY_MEMBER` account type a while ago, so you don't need this special switch case anymore

File: src/PageContent/Directory/DirectoryPage.js [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+        // keeps listening for real-time database change 
+        db.ref('accounts').orderByChild('umbrella').equalTo(this.state.umbrellaId).on('value', (snapshot) => { 
+            let accountObjects = snapshot.val();       
+            let raList = [];
+            let dgList = [];
+
+            // loop through all AccountObjects
+            for (let key in accountObjects) {
+                let accountItem = accountObjects[key];
+                if (accountObjects.hasOwnProperty(key)  // filter out prototype props
```
Comment (arielin3) [inline src/PageContent/Directory/DirectoryPage.js]: what does this do?

Comment (leontaolong) [reply src/PageContent/Directory/DirectoryPage.js]: It loops through each object in `accountObjects`. `accountObjects.hasOwnProperty(key` is to filter out `prototype props` (if there's any) of an javascript object.

File: src/PageContent/Directory/DirectoryPage.js (hunk 2 above)
Comment (leontaolong) [reply src/PageContent/Directory/DirectoryPage.js]: Updated!

**Pull request merged**